cfme/configure/tasks.py

```diff
@@ -105,6 +109,167 @@ def is_analysis_finished(name, task_type='vm', clear_tasks_after_success=True):
                             clear_tasks_after_success=clear_tasks_after_success)
 
 
+def are_all_tasks_finished(name, expected_num_of_tasks, task_type='container'):
+    """ Check if all tasks states are finished - if not, reload page"""
+
+    tabs_data = {
+        'container': {
+            'tab': 'AllTasks',
+            'task': '{}',
+            'state': 'finished'
+        },
+        'vm': {
+            'tab': 'AllTasks',
+            'task': 'Scan from Vm {}',
+            'state': 'finished'
+        },
+        'host': {
+            'tab': 'MyOtherTasks',
+            'task': "SmartState Analysis for '{}'",
+            'state': 'Finished'
+        },
+        'datastore': {
+            'tab': 'MyOtherTasks',
+            'task': 'SmartState Analysis for [{}]',
+            'state': "Finished"
+        },
+        'cluster': {
+            'tab': 'MyOtherTasks',
+            'task': 'SmartState Analysis for [{}]',
+            'state': "Finished"
+        }
+    }[task_type]
+    return all_tasks_finished(
+        destination=tabs_data['tab'],
+        task_name=tabs_data['task'].format(name),
+        expected_status=tabs_data['state'],
+        expected_num_of_tasks=expected_num_of_tasks
+    )
+
+
+def all_tasks_finished(destination, task_name, expected_status, expected_num_of_tasks):
```

Comment:
**Required**

This method is 'all_tasks_finished', but it takes an expected status, and thus is really an 'all_tasks_match_status' method.

Rename the method, or don't take an expected status.  You're already using re.IGNORECASE, when the only difference between the states in tab_data is a capital F.

Reply:
Done!!!
Rename "all_tasks_finished" to "all_tasks_match_status"
And "are_all_tasks_finished" to "are_all_tasks_match_status"

I use the expected_status in order to filter rows in Tasks table.
The Capital F is change from appliance build version, 
Using  re.IGNORECASE prevent the need to change it all the time on TABS_DATA_PER_PROVIDER.